user-guide/Advanced_Modules/DOM/Tutorials/DOM_Remove_Enum_Entry.md

```diff
@@ -0,0 +1,406 @@
+
+            var filter = SectionDefinitionExposers.ID.Equal(sectionDefinitionId);
+
+            var sectionDefinition = _domHelper.SectionDefinitions.Read(filter).First();
+            var fieldDescriptor = sectionDefinition.GetFieldDescriptorById(fieldDescriptorId) as GenericEnumFieldDescriptor;
+
+            var enumDelete = fieldDescriptor.GenericEnumInstance.Entries.First(x => x.DisplayName == enumName);
+            enumDelete.IsSoftDeleted = true;
+            _domHelper.SectionDefinitions.Update(sectionDefinition);
+        }
+    }
+}
+
+```
+
+After soft-deletion, you should not see that enum entry in the dropdown of the form on your low code app.
+
+   ![Step2](~/user-guide/images/DOM_Remove_Enum_Entry_Step2.png)
+
+## Step 3: Remove the value from existing instances
+
+The given code does the following:
+1. Gets all instances that use the enum entry.
+1. Removes the enum value.
+1. Updates the instances.
+
+```C#
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Skyline.DataMiner.Automation;
+using Skyline.DataMiner.Net.Apps.DataMinerObjectModel;
+using Skyline.DataMiner.Net.Apps.Modules;
+using Skyline.DataMiner.Net.Apps.Sections.SectionDefinitions;
+using Skyline.DataMiner.Net.GenericEnums;
+using Skyline.DataMiner.Net.ManagerStore;
+using Skyline.DataMiner.Net.Messages.SLDataGateway;
+using Skyline.DataMiner.Net.Sections;
+
+
+namespace Tutorial
+{
+    ///<summary>
+    /// This script will remove the enum value "0" (int value of "Director") from all existing instances.
+    ///</summary>
+    public class Script
+    {
+        private const string ModuleName = "people_app";
+        private DomHelper _domHelper;
+        private Engine _engine;
+
+        public void Run(Engine engine)
+        {
+            _engine = engine;
+            _domHelper = new DomHelper(_engine.SendSLNetMessages, ModuleName);
+
+            var sectionDefinitionName = "PeopleSectionDefinition";
+            var fieldDescriptorName = "JobTitle";
+            var enumValue = 0; // int value of the enum entry "Director" we want to delete
+
+            var sectionDefinitions = _domHelper.SectionDefinitions.ReadAll();
+            var sectionDefinition = sectionDefinitions.First(x => x.GetName() == sectionDefinitionName);
+            var fieldDescriptor = sectionDefinition.GetAllFieldDescriptors().First(x => x.Name == fieldDescriptorName) as GenericEnumFieldDescriptor;
+            if(fieldDescriptor == null){return;}
+
+            var filter = DomInstanceExposers.FieldValues.DomInstanceField(fieldDescriptor.ID).Contains(enumValue);
+            var instancesToDelete = _domHelper.DomInstances.Read(filter);
+
+            foreach (var instance in instancesToDelete)
+            {
+                var section = instance.Sections.First(x => x.SectionDefinitionID.Equals(sectionDefinition.GetID()));
+                section.RemoveFieldValueById(fieldDescriptor.ID);
+                _domHelper.DomInstances.Update(instance);
```

Comment:
Isn't there a method that can be used to remove a FieldValue from a DomInstance without needing to search for the Section ourselves?

Reply:
There is no method to remove a FieldValue on a DomInstance